requirements.txt

```diff
@@ -1,6 +1,103 @@
 #
-# This file is autogenerated by pip-compile with python 3.10
-# To update, run:
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
 #
 #    pip-compile requirements.in
 #
+astroid==2.14.2
+    # via pylint
+attrs==22.2.0
+    # via pytest
+black==23.1.0
+    # via -r requirements.in
+build==0.10.0
+    # via pip-tools
+cfgv==3.3.1
+    # via pre-commit
+click==8.1.3
+    # via
+    #   black
+    #   pip-tools
+coverage==7.2.0
+    # via -r requirements.in
+dill==0.3.6
+    # via pylint
+distlib==0.3.6
+    # via virtualenv
+exceptiongroup==1.1.0
+    # via pytest
+filelock==3.9.0
+    # via virtualenv
+flake8==6.0.0
+    # via -r requirements.in
+identify==2.5.18
+    # via pre-commit
+iniconfig==2.0.0
+    # via pytest
+isort==5.12.0
+    # via pylint
+lazy-object-proxy==1.9.0
+    # via astroid
+mccabe==0.7.0
+    # via
+    #   flake8
+    #   pylint
+mypy-extensions==1.0.0
+    # via black
+nodeenv==1.7.0
+    # via pre-commit
+packaging==23.0
+    # via
+    #   black
+    #   build
+    #   pytest
+pathspec==0.11.0
+    # via black
+pip-tools==6.12.2
+    # via -r requirements.in
+platformdirs==3.0.0
+    # via
+    #   black
+    #   pylint
+    #   virtualenv
+pluggy==1.0.0
+    # via pytest
+pre-commit==3.1.0
+    # via -r requirements.in
+pycodestyle==2.10.0
+    # via flake8
+pyflakes==3.0.1
+    # via flake8
+pylint==2.16.2
+    # via -r requirements.in
+pyproject-hooks==1.0.0
+    # via build
+pytest==7.2.1
+    # via
+    #   -r requirements.in
+    #   pytest-mock
+pytest-mock==3.10.0
+    # via -r requirements.in
+pyyaml==6.0
+    # via pre-commit
+tomli==2.0.1
+    # via
+    #   black
+    #   build
+    #   pylint
+    #   pytest
+tomlkit==0.11.6
+    # via pylint
+typing-extensions==4.5.0
+    # via astroid
+virtualenv==20.19.0
+    # via pre-commit
+wheel==0.38.4
+    # via pip-tools
+wrapt==1.14.1
+    # via astroid
+
+# The following packages are considered to be unsafe in a requirements file:
+# pip
+# setuptools
+setuptools>=65.5.1 # not directly required, pinned by Snyk to avoid a vulnerability
```

Comment:
This one specifically is interesting, so I think we should generate the `requirements.txt` still even with the empty `requirements.in` as I mentioned.

Reply:
The Snyk checks failed without this. I do agree we should have minimum and maximum package version numbers to protect against vulnerabilities and guarantee success. That is a good best practice.